frontend/src/metabase/metadata/components/SortableFieldList/SortableFieldList.tsx

```diff
@@ -0,0 +1,52 @@
+import { PointerSensor, useSensor } from "@dnd-kit/core";
+import { useMemo } from "react";
+
+import {
+  type DragEndEvent,
+  SortableList,
+} from "metabase/core/components/Sortable";
+import { Flex } from "metabase/ui";
+import type { Table } from "metabase-types/api";
+
+import { SortableField } from "../SortableField";
+
+import { getId, getItems, getItemsOrder, sortItems } from "./utils";
+
+interface Props {
+  table: Table;
+  onChange: (fieldOrder: DragEndEvent["itemIds"]) => void;
+}
+
+export const SortableFieldList = ({ table, onChange }: Props) => {
+  const pointerSensor = useSensor(PointerSensor, {
+    activationConstraint: { distance: 15 },
+  });
+  const items = useMemo(() => getItems(table), [table]);
+  const order = useMemo(() => getItemsOrder(items), [items]);
+  const sortedItems = useMemo(() => sortItems(items, order), [items, order]);
+  const isDragDisabled = sortedItems.length <= 1;
+
+  const handleSortEnd = ({ itemIds }: DragEndEvent) => {
+    onChange(itemIds);
+  };
+
+  return (
+    <Flex direction="column" gap="sm">
+      <SortableList
```

Comment:
AFAICT this component is not virtualized. Will we run into problems with huge tables?

(this is not a blocker for the PR).

Reply:
Details in this thread: https://metaboat.slack.com/archives/C01MS7DQKR6/p1745490386637529 fr